### PR TITLE
Parsing command line options, fixes #27

### LINKED ISSUE #27: Jar file arguments
At the moment, if you want to run the project using the cubelets, you must pass the location of the front end to serve before the location of the cubelet program.

This is not ideal. Here are some options:

1. Leave it as it is: to use the cubelets you have to pass the front end location to serve.
2. Make it so that you never pass the front end to serve, always serve it from within the jar.
3. Use command line flags: probably assume serve front end from jar but if a flag is given then serve from given location.

I think 3 is probably best - thoughts?

### PR DESCRIPTION
**Pull request by KoviRobi**
Using apache commons cli 1.2

### CODE REVIEW

File: java-backend/nativeServer/src/main/java/com/github/cambridgeAlphaTeam/Server.java [hunk 1]
```diff
@@ -34,8 +41,56 @@
...
+         *           String description)
+         *
+         * where a short option is like "-h" and a long option is like
+         * "--help" (which takes no arguments). The description is for
+         * generating the help message.
+         */
+        options.addOption ("h", "help", false, "Print this help message");
+        options.addOption ("u", "usage", false, "Print the list of options");
+        options.addOption ("c", "cubelets-program", true, "A single string of the cubelet program. " +
+                "If you want to pass arguments to it, make a batch/shell file and put the arguments to that file, " +
```
Comment (ird28) [inline java-backend/nativeServer/src/main/java/com/github/cambridgeAlphaTeam/Server.java]: Is this the best way of passing the arguments to the cubelet program? What's the problem with having command line arguments?

Comment (KoviRobi) [inline java-backend/nativeServer/src/main/java/com/github/cambridgeAlphaTeam/Server.java]: How do you delimit those, i.e. how would the parser know when to stop parsing it as cubelet options.

Comment (KoviRobi) [inline java-backend/nativeServer/src/main/java/com/github/cambridgeAlphaTeam/Server.java]: Anyway, I am using getOptionValues(), but unfortunately it is a bit underdocumented:
https://commons.apache.org/proper/commons-cli/javadocs/api-release/org/apache/commons/cli/CommandLine.html#getOptionValues(char)

Comment (KoviRobi) [inline java-backend/nativeServer/src/main/java/com/github/cambridgeAlphaTeam/Server.java]: Ok, figured out what it does: we can use "-c prog -c arg1 -c arg2", will document it now.

**Pull request merged**